compiler/test/fail_compilation/placenew.d

```diff
@@ -0,0 +1,80 @@
+/* TEST_OUTPUT:
+---
+fail_compilation/placenew.d(23): Error: PlacementExpression `3` is an rvalue, but must be an lvalue
+fail_compilation/placenew.d(28): Error: undefined identifier `x`
+fail_compilation/placenew.d(36): Error: `new ( i )` PlacementExpression cannot be evaluated at compile time
+fail_compilation/placenew.d(39):        called from here: `xxx()`
+fail_compilation/placenew.d(39):        while evaluating: `static assert(xxx() == 1)`
+fail_compilation/placenew.d(48): Error: new placement size 24 must be >= object size 40
+fail_compilation/placenew.d(54): Error: placement new cannot be used with associative arrays
+fail_compilation/placenew.d(67): Error: new placement size 4 must be >= class object size $?:32=16|64=24$
+fail_compilation/placenew.d(77): Error: `@safe` function `test7` cannot use placement `new`
+---
+*/
+
+void test0()
+{
+    int i;
+    int* pi = new (i) int;
```

Comment:
I thought we ejected this pattern into space? Please, just no.

Reply:
I responded to this in the other comment.